packages/babel-plugin-transform-object-super/src/index.ts

```diff
@@ -37,7 +37,11 @@ export default declare(api => {
         });
 
         if (objectRef) {
-          path.scope.push({ id: t.cloneNode(objectRef) });
+          const scopePath = path.scope.getBlockParent().path;
+          path.scope.push({
+            id: t.cloneNode(objectRef),
+            kind: scopePath.parentPath?.isLoop() ? "let" : "var",
+          });
```

Comment:
Checking direct parent is not sufficient. For example, wrap the loop body in `if (true) { ... }` and you get a `var` again.

Reply:
Fixed, thank you!